tests/util/pilot_server.go

```diff
@@ -77,39 +77,40 @@ func setup(additionalArgs ...func(*bootstrap.PilotArgs)) (*bootstrap.Server, Tea
 
 	meshConfig := mesh.DefaultMeshConfig()
 	// Create a test pilot discovery service configured to watch the tempDir.
-	args := bootstrap.PilotArgs{
-		Namespace: "testing",
-		DiscoveryOptions: bootstrap.DiscoveryServiceOptions{
+	args := bootstrap.NewPilotArgs(func(p *bootstrap.PilotArgs) {
+		p.Namespace = "testing"
+		p.DiscoveryOptions = bootstrap.DiscoveryServiceOptions{
 			HTTPAddr:        httpAddr,
 			GrpcAddr:        ":0",
 			SecureGrpcAddr:  ":0",
 			EnableProfiling: true,
-		},
+		}
 		//TODO: start mixer first, get its address
-		Mesh: bootstrap.MeshArgs{
+		p.Mesh = bootstrap.MeshArgs{
 			MixerAddress: "istio-mixer.istio-system:9091",
-		},
-		Config: bootstrap.ConfigArgs{
+		}
+		p.Config = bootstrap.ConfigArgs{
 			KubeConfig: env.IstioSrc + "/tests/util/kubeconfig",
-		},
-		MeshConfig:        &meshConfig,
-		MCPMaxMessageSize: 1024 * 1024 * 4,
-		KeepaliveOptions:  keepalive.DefaultOption(),
-		ForceStop:         true,
+		}
+		p.MeshConfig = &meshConfig
+		p.MCPMaxMessageSize = 1024 * 1024 * 4
+		p.KeepaliveOptions = keepalive.DefaultOption()
+		p.ForceStop = true
 		// TODO: add the plugins, so local tests are closer to reality and test full generation
 		// Plugins:           bootstrap.DefaultPlugins,
-	}
+	})
+
```

Comment:
Shouldn't `args.COnfig.FileDir` be set in the lambda above?

Reply:
Yes. It needs to be. Thanks, Changed it now.